### DISCUSSION TITLE
[프로그래머스] 스터디 코딩 문제(5문제)_2주차

### DISCUSSION THREAD

**Discussion by kth232** [soeunl/502-3-Study#29] [General]
1. [입문] 중앙값 구하기
    
    중앙값은 어떤 주어진 값들을 크기의 순서대로 정렬했을 때 가장 중앙에 위치하는 값을 의미합니다. 예를 들어 1, 2, 7, 10, 11의 중앙값은 7입니다. 정수 배열 array가 매개변수로 주어질 때, 중앙값을 return 하도록 solution 함수를 완성해보세요.
    
    단,  array의 길이는 무조건 홀수로 제한합니다.
    
    https://school.programmers.co.kr/learn/courses/30/lessons/120811
    
2. [입문] 아이스 아메리카노
    
    아이스 아메리카노는 한잔에 5,500원입니다. 머쓱이가 가지고 있는 돈 money가 매개변수로 주어질 때, 머쓱이가 최대로 마실 수 있는 아메리카노의 잔 수와 남는 돈을 순서대로 담은 배열을 return 하도록 solution 함수를 완성해보세요.
    
    https://school.programmers.co.kr/learn/courses/30/lessons/120819?language=java
    
3. [입문] 중복된 숫자 개수
    
    정수가 담긴 배열 array와 정수 n이 매개변수로 주어질 때, array에 n이 몇 개 있는 지를 return 하도록 solution 함수를 완성해보세요.
    
    https://school.programmers.co.kr/learn/courses/30/lessons/120583  

    
4. [입문] 옷 가게 할인 받기
    
    머쓱이네 옷가게는 10만 원 이상 사면 5%, 30만 원 이상 사면 10%, 50만 원 이상 사면 20%를 할인해줍니다.
    구매한 옷의 가격 price가 주어질 때, 지불해야 할 금액을 return 하도록 solution 함수를 완성해보세요.
    
    https://school.programmers.co.kr/learn/courses/30/lessons/120818      

5. [기초] 할일 목록
    
    오늘 해야 할 일이 담긴 문자열 배열 todo_list와 각각의 일을 지금 마쳤는지를 나타내는 boolean 배열 finished가 매개변수로 주어질 때, todo_list에서 아직 마치지 못한 일들을 순서대로 담은 문자열 배열을 return 하는 solution 함수를 작성해 주세요.
    
    단, 아직 마치지 못한 일이 최소 1개는 있습니다.
    
    https://school.programmers.co.kr/learn/courses/30/lessons/181885

**Comment by heyejiyang** [comment]
1.
```
import java.util.Arrays;

class Solution {
    public int solution(int[] array) {
        Arrays.sort(array);
        return array[array.length / 2];
    }
}
```
2.
```
import java.util.Arrays;
class Solution {
    public int[] solution(int money) {
        int cup = (money/5500);
        int change = money - cup * 5500;
        int[] answer = {cup, change};
        return answer;
    }
}
```
3.
```
class Solution {
    public int solution(int[] array, int n) {
        int answer = 0;
        for(int i : array){
            if(i == n) answer++;
        }
        return answer;
    }
}
```
4.
```
class Solution {
    public int solution(int price) {
        int answer = 0;
       if(price >= 500000){
            answer = (int)(price*0.8);
        }else if(price >= 300000){
            answer = (int)(price*0.9);
        }else if(price >= 100000){
            answer = (int)(price*0.95);
        }else{
            answer = price;
        }
        return answer;
    }
}
```
5.
```
import java.util.ArrayList;
import java.util.Arrays;
class Solution {
    public String[] solution(String[] todo_list, boolean[] finished) {
        
        ArrayList<String> unfinishedTasks = new ArrayList<>();
        
        for(int i = 0; i< finished.length; i++){
                if(!finished[i]){
                     unfinishedTasks.add(todo_list[i]);
                }
        }
         String[] answer = unfinishedTasks.toArray(new String[0]);
        
        return answer;
    }
}
```

**Reply by heyejiyang** [reply]
![image](https://github.com/soeunl/502-3-Study/assets/132132524/b7266505-2eae-4179-8b4b-40b6cec793b1)

**Reply by heyejiyang** [reply]
toArray() 메서드는 ArrayList의 내부 크기에 따라 필요에 따라 새로운 배열을 생성하거나, 이미 충분한 크기의 배열이 주어진 경우 그 배열을 사용하여 요소들을 복사한 후 반환합니다.

**Comment by hyeon12** [comment]
1. [입문] 중앙값 구하기
```
import java.util.Arrays;

class Solution {
    public int solution(int[] array) {
        Arrays.sort(array);
        int center = array.length / 2;
        return array[center];
    }
}
```
*import

2. [입문] 아이스 아메리카노
```
class Solution {
    public int[] solution(int money) {
        int americano = money/5500;
        int change = money%5500;
        int[] answer = {americano, change};
        return answer;
    }
}
```

3. [입문] 중복된 숫자 개수
```
class Solution {
    public int solution(int[] array, int n) {
        int target = 0;
        
        for(int i : array){
            if(i == n){
            target++;}
        }
        
        return target;
    }
}
```

4. [입문] 옷 가게 할인 받기
```
class Solution {
    public int solution(int price) {
        
        int pay = 0;
        if (price >= 500000) {
            pay = (int)(price - (price * 0.2));
        } else if (price >= 300000) {
            pay = (int)(price - (price * 0.1));
        } else if (price >= 100000) {
            pay = (int)(price - (price * 0.05));
        } else {
            pay = price;
        }
        return pay;
    }
}
```
* 할인을 받지 못하는 상황도 고려해야 합니다!!

5.[기초] 할일 목록

**Comment by siwon9** [comment]
Q1
```
package Question;

import java.util.Arrays;

public class Q1 {
    public static void main(String[] args) {
        int array[] = {1,3,11,8,9};
        System.out.println(solution(array));
    }

    public static int solution(int[] array){
        Arrays.sort(array);
        int middle = array.length/2;

        if(array.length % 2 == 0){
            System.out.println("배열을 홀수단위로 입력하세요");
            return 0;
        }

        return array[middle];
    }
}

```
Q2
```
class Solution {
    public int[] solution(int money) {
        int[] answer = new int[2];

        int cal = money / 5500;
        int left = money - 5500 * cal;
        
        answer[0] = cal; 
        answer[1] = left;     

        return answer;
    }
}
```
Q3
```
    public static int solution(int[] array, int n) {
        int count = 0;
        for(int i : array) {
            if(i==n) {
                count++;
            }
        }
        return count;
    }
```
Q4.
```
package Question;

public class Q4 {
    public static void main(String[] args) {
        int price = 150000;
        int result = solution(price);
        System.out.printf("총 결재금액은 %d원 입니다.", result);
}
    public static int solution(int price) {
        if (price >= 100000 && price < 300000) {
            price -= price * 0.05;
        } else if (price >= 300000 && price < 500000) {
            price -= price * 0.1;
        } else if (price >= 500000) {
            price -= price * 0.2;
        } else {
            price = price;
        }
        return price;
    }
}
```
Q5
```

```

**Comment by soeunl** [comment]
Q1
![Quiz01](https://github.com/soeunl/502-3-Study/assets/154653812/07de62e8-73ba-46af-8fd7-794ecc961492)
<br>
Q2
![Quiz02](https://github.com/soeunl/502-3-Study/assets/154653812/c915539d-35b3-4954-ab06-fecc14e46baf)
<br>
Q3
![Quiz03](https://github.com/soeunl/502-3-Study/assets/154653812/a64709a9-be1c-4927-976c-a28e65090071)
<br>
Q4
![Quiz04](https://github.com/soeunl/502-3-Study/assets/154653812/70b07098-7573-4479-9467-3748150054a0)
<br>
Q5
![Quiz05](https://github.com/soeunl/502-3-Study/assets/154653812/330796f1-2fc6-4a6b-90dc-796ee311ed0b)
5번 문제는 오류났습니다ㅠㅠ unfinished라는 빈 배열을 만들어놓고, todo_list 값 중에 finished가 아닌 값이 있으면 unfinished 배열에 넣어서 출력하고 싶었는데, 잘 안됩니닷

**Reply by heyejiyang** [reply]
저도 배열에 넣고싶어서... 방법을 gpt한테 물어봤답니다ㅜ.ㅜ 배열로 만들면 복잡해서? 리스트 쓰라고 하더라구요.....

**Comment by beom33** [comment]
Q1. 
    
class Solution {
    public int solution(int[] array) {
         Arrays.sort(array);
        int center = arrays.length;
        int answer = 0;
        return answer;
    }
}
         
Q2.

class Solution {
    public int[] solution(int money) {
        int americano = money/5500;
        int change = money - americano*5500
        int[] answer = {americano, change};
        return answer;
    }
}




Q3.


class Solution {
    public int solution(int[] array, int n) {
          int answer : 0;
          ...
        int answer = 0;
        return answer;
    }
}
